File: ai-career-frontend/services/subscription.ts
import  api  from "./api";

export interface Subscription {
  id: number;
  user_id: number;
  plan_name: string;
  price: number;
  status: string;
  start_date: string;
  end_date?: string | null;
}

export interface Usage {
  resume_analysis_count: number;
  interview_count: number;
  question_count: number;
  voice_interview_count: number;
}

export interface SubscriptionStatus {
  subscription: Subscription;
  usage: Usage;
  is_premium: boolean;
}

export async function getSubscriptionStatus() {
  const response =
    await api.get<SubscriptionStatus>(
      "/subscriptions/status"
    );

  return response.data;
}

File: backend/schemas/subscription.py
from datetime import datetime
from pydantic import BaseModel


class PlanResponse(BaseModel):
    name: str
    price: float
    duration_days: int
    features: list[str]


class SubscriptionResponse(BaseModel):
    id: int
    user_id: int
    plan_name: str
    price: float
    status: str
    start_date: datetime
    end_date: datetime | None = None

    class Config:
        from_attributes = True


class UsageResponse(BaseModel):
    resume_analysis_count: int
    interview_count: int
    question_count: int
    voice_interview_count: int
    
class SubscriptionStatusResponse(BaseModel):
    subscription: SubscriptionResponse
    usage: UsageResponse
    is_premium: bool